controllers/tempo/microservices_controller.go

```diff
@@ -132,13 +142,32 @@ func (r *MicroservicesReconciler) getStorageConfig(ctx context.Context, tempo v1
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *MicroservicesReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Add an index to the storage secret field in the Microservices CRD.
+	// If the content of any secret changes, the watcher can identify related Microservices CRs
+	// and reconcile them (i.e. update the tempo configuration file and restart the pods)
+	err := mgr.GetFieldIndexer().IndexField(context.Background(), &v1alpha1.Microservices{}, storageSecretField, func(rawObj client.Object) []string {
+		microservices := rawObj.(*v1alpha1.Microservices)
+		if microservices.Spec.Storage.Secret == "" {
+			return nil
+		}
+		return []string{microservices.Spec.Storage.Secret}
+	})
+	if err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Microservices{}).
 		Owns(&corev1.ConfigMap{}).
 		Owns(&corev1.ServiceAccount{}).
 		Owns(&corev1.Service{}).
 		Owns(&v1.StatefulSet{}).
 		Owns(&v1.Deployment{}).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
```

Comment:
I'm not 100% happy that this code watches the entire cluster for every change to any secret, but afaics that's how it's supposed to work when the operator needs to react on changes of external resources: https://book.kubebuilder.io/reference/watching-resources/externally-managed.html